static/app/components/modals/inviteMembersModal/index.spec.tsx

```diff
@@ -159,120 +187,154 @@ describe('InviteMembersModal', function () {
+      `/organizations/org-slug/members/`,
       expect.objectContaining({
         data: {email: 'test2@test.com', role: 'member', teams: []},
       })
     );
+  });
+
+  it('can reset modal', async function () {
+    setupView({
+      mockApiResponses: [defaultMockOrganizationRoles, defaultMockPostOrganizationMember],
+    });
+    expect(await screen.findByRole('button', {name: 'Add another'})).toBeInTheDocument();
+
+    await setupMemberInviteState();
+    await userEvent.click(screen.getByRole('button', {name: 'Send invites (2)'}));
 
     // Wait for them to finish
     expect(
       await screen.findByText(textWithMarkupMatcher('Sent 2 invites'))
     ).toBeInTheDocument();
 
-    expect(screen.getByRole('button', {name: 'Close'})).toBeInTheDocument();
-    expect(screen.getByRole('button', {name: 'Send more invites'})).toBeInTheDocument();
-
-    // Send more reset the modal
+    // Reset the modal
     await userEvent.click(screen.getByRole('button', {name: 'Send more invites'}));
-
     expect(screen.getByRole('button', {name: 'Send invite'})).toBeDisabled();
   });
 
-  it('marks failed invites', async function () {
-    jest.mocked(useOrganization).mockReturnValue(org);
-    const faildCreateMemberMock = MockApiClient.addMockResponse({
-      url: `/organizations/${org.slug}/members/`,
-      method: 'POST',
-      statusCode: 400,
+  it('sends all successful invites with team default', async function () {
+    const {mocks} = setupView({
+      mockApiResponses: [defaultMockOrganizationRoles, defaultMockPostOrganizationMember],
     });
 
-    render(<InviteMembersModal {...modalProps} />);
+    expect(await screen.findByRole('button', {name: 'Add another'})).toBeInTheDocument();
+    await setupMemberInviteState();
 
-    await waitFor(() => {
-      expect(screen.getByRole('textbox', {name: 'Email Addresses'})).toBeInTheDocument();
-    });
+    await userEvent.click(screen.getByRole('button', {name: 'Send invites (2)'}));
 
+    const mockPostApi = mocks[1];
+    expect(mockPostApi).toHaveBeenCalledTimes(2);
+    expect(mockPostApi).toHaveBeenNthCalledWith(
+      1,
+      `/organizations/org-slug/members/`,
+      expect.objectContaining({
+        data: {email: 'test1@test.com', role: 'admin', teams: ['team-slug']},
+      })
+    );
+    expect(mockPostApi).toHaveBeenNthCalledWith(
+      2,
+      `/organizations/org-slug/members/`,
+      expect.objectContaining({
+        data: {email: 'test2@test.com', role: 'member', teams: ['team-slug']},
+      })
+    );
+  });
+
+  it('does not use defaults when there are multiple teams', async function () {
+    const another_team = TeamFixture({id: '2', slug: 'team2'});
+    setupView({orgTeams: [TeamFixture(), another_team]});
+
+    expect(await screen.findByRole('button', {name: 'Add another'})).toBeInTheDocument();
+
+    await userEvent.click(screen.getByRole('button', {name: 'Add another'}));
+    const teamInputs = screen.getAllByRole('textbox', {name: 'Add to Team'});
```

Comment:
nit using `findBy` selectors is a little safer since they're async. for this case, if something changes when a user clicks we may need to wait for a render cycle.

```suggestion
    const teamInputs = await screen.findAllByRole('textbox', {name: 'Add to Team'});
```

Reply:
Good catch. Let me pull this into a follow up PR so the changes are easier to reivew